### PR TITLE
allow to specify status for UNAUTHTORIZED

### LINKED ISSUE #18: Make URL harvesting impossible
At the moment it is possible to discover URLs of the protected services.

When client sends a request without a token (or with invalid token), OpenTAXII doesn't know if authentication is required for the service, so first it has to configure service instance. If a service does not exist, `404 NOT FOUND` will be returned. if service is configured and requires authentication, `UNAUTHORIZED` will be returned. 

So it is possible to walk all possible URLs and log the ones with `UNAUTHORIZED`. This is a configuration leak.

### PR DESCRIPTION
**Pull request by orsinium**
Fix #18 with backward compatibility. Allow specifying alternative TAXII status type instead of "UNAUTHTORIZED". By default, use unauthorized how it recommend by spec (5.1 HTTP Responses as TAXII Status Messages).

### CODE REVIEW

File: opentaxii/defaults.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ save_raw_inbox_messages: yes
 xml_parser_supports_huge_tree: yes
 count_blocks_in_poll_responses: no
 return_server_error_details: no
+unauthorized_status: UNAUTHORIZED
```
Comment (andreasEIQ) [inline opentaxii/defaults.yml]: could we add a comment explaining what options exist for this setting?

File: opentaxii/middleware.py [hunk 2]
```diff
@@ -73,7 +73,9 @@ def wrapper(relative_path=""):
             if service:
                 if (service.authentication_required
                         and context.account is None):
-                    raise UnauthorizedException()
+                    raise UnauthorizedException(
+                        status_type=server.config['unauthorized_status'],
```
Comment (andreasEIQ) [inline opentaxii/middleware.py]: it's unclear to me what values are accepted for `status_type`, do you have a source for this information?

File: opentaxii/taxii/exceptions.py [hunk 3]
```diff
@@ -36,8 +36,8 @@ def __init__(self, message, **kwargs):
 
 class UnauthorizedStatus(StatusMessageException):
 
-    def __init__(self, **kwargs):
-        super(UnauthorizedStatus, self).__init__(ST_UNAUTHORIZED, **kwargs)
+    def __init__(self, status_type=ST_UNAUTHORIZED, **kwargs):
+        super(UnauthorizedStatus, self).__init__(status_type=status_type, **kwargs)
```
Comment (andreasEIQ) [inline opentaxii/taxii/exceptions.py]: seems to me like we should just remove this class if it's possible to pass status types other than "unauthorized"

File: opentaxii/middleware.py (hunk 2 above)
Comment (orsinium) [reply opentaxii/middleware.py]: Yes, `libtaxii.constants.ST_TYPES_11`. I've updated the docs.

File: opentaxii/defaults.yml (hunk 1 above)
Comment (orsinium) [reply opentaxii/defaults.yml]: I've added validation into Config reader and updated docs :+1:

File: opentaxii/taxii/exceptions.py (hunk 3 above)
Comment (orsinium) [reply opentaxii/taxii/exceptions.py]: We can't just remove intermediate exceptions without breaking backward compatibility. What if someone doing `except UnauthorizedStatus:`?

I think, there is nothing bad in having `UnauthorizedStatus` with `status_type="NOT_FOUND"`, for example. `status_type` is about the response, exception type is about internal server handling.

Comment (andreasEIQ) [reply opentaxii/taxii/exceptions.py]: > We can't just remove intermediate exceptions without breaking backward compatibility. What if someone doing except UnauthorizedStatus:?

This is a good reason, although I think OpenTAXII should be provided as-is and not be used as a library.

Comment (orsinium) [reply opentaxii/taxii/exceptions.py]: Good point. Moved into #150

**Pull request merged**